java/org.travlyn.server/src/main/java/org/travlyn/server/externalapi/access/OpenRouteDirectionRequest.java

```diff
@@ -0,0 +1,92 @@
+package org.travlyn.server.externalapi.access;
+
+import com.google.gson.*;
+import org.travlyn.server.util.Pair;
+import org.travlyn.shared.model.api.ExecutionInfo;
+import org.travlyn.shared.model.api.Step;
+import org.travlyn.shared.model.api.Stop;
+import org.travlyn.shared.model.api.Waypoint;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+public abstract class OpenRouteDirectionRequest implements Request<ExecutionInfo> {
+    private static final String PROFILE = "foot-walking";
+    private static final String BASE_URL = "https://api.openrouteservice.org/v2/directions/"+PROFILE+"/geojson";
+    private final Gson gson = new Gson();
+
+    @Override
+    public abstract ExecutionInfo getResult();
+
+    protected JsonObject makeAPICall(List<Pair<Double,Double>> wayPoints){
```

Comment:
Maybe add optional parameter for instruction language, that is given by `Accept-Language`-Header